Clarify that init callback can't be used from backend code and why (#5645)
slint-ui/slint: docs/reference/src/language/builtins/callbacks.md

```

Don't use this callback to initialize properties, because this violates the declarative principle.

> Even though the `init` callback also exists on top-level, exported components (e.g. `Window`), it
> cannot be bound to backend code (i.e. an `on_init` function does not exist in the generated code).
> This is because the callback is invoked during the creation of the component, before you could
> call `on_init` to actually set it.

Reviewer: ```suggestion
Even though the `init` callback exists on all components, it cannot be set from application code,
i.e. an `on_init` function does not exist in the generated code. This is because the callback is invoked during the creation of the component, before you could call `on_init` to actually set it.
```
